language/move-compiler/src/expansion/aliases.rs

```diff
@@ -152,28 +153,51 @@ impl AliasMap {
         self.unused.len()
     }
 
-    pub fn module_alias_get(&mut self, n: &Name) -> Option<&ModuleIdent> {
+    pub fn module_alias_get(&mut self, n: &Name) -> Option<ModuleIdent> {
         match self.modules.get_mut(n) {
             None => None,
             Some((depth_opt, ident)) => {
                 if let Some(depth) = depth_opt {
                     self.unused[*depth].modules.remove(n);
                 }
                 *depth_opt = None;
-                Some(ident)
+                // What we are trying to do here is to preserve the
+                // location of the module name rather than replacing
+                // it with the location of the alias. The n parameter
+                // represents JUST the module name, so we leave
+                // locations related to module's address intact as
+                // replacing them with the location of the module name
+                // does not seem right
```

Comment:
Maybe just 
```
// We are preserving the names original location, rather than referring to where the alias was defined. 
// TODO maybe we should also keep the aliases loc? Or some reference to the alias?
```
I like the addition of a comment, but it feels just a tad long